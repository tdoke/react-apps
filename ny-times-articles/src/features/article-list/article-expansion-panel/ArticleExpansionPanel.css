.article-expansion-panel-container {
  box-sizing: border-box;
  background-color: #FAFAFA;
  margin-top: 2rem;
  font-size: 0.9rem;
}

.article-expansion-panel-container:hover {
  background-color: lightgray;
  cursor: pointer;
}

.article-expansion-panel-container:nth-of-type(1) {
  margin: 0;
  padding-top: 2rem;
}

.article-expansion-panel-container .container__centerItems {
  display: flex;
  justify-content: center;
  align-items: center;
}

.article-expansion-panel-container p {
  margin: 0;
}

.article-expansion-panel-container .published-date {
  margin-top: 0.5rem;
}

.article-expansion-panel-container .article-by {
  margin-top: 0.5rem;
  color: #585858;
}