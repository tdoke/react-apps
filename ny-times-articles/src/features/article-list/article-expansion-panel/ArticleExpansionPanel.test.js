import React from 'react';
import { shallow } from 'enzyme';
import ArticleExpansionPanel from './ArticleExpansionPanel.component';

describe('ArticleExpansionPanel', () => {
  let props;
  let getComponent
  beforeEach(() => {
    props = {
      article: {
        title: 'world cup 2019',
        byline: 'Harsha Bhogle',
        url: 'https://www.cricbuzz.com',
        published_date: '2019/05/05'
      },
      onArticleClick: jest.fn()
    }
    getComponent = () => shallow(<ArticleExpansionPanel {...props} />);
  });

  it('should render ArticleExpansionPanel', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should call onArticleClick, on click', () => {
    const component = getComponent();
    component
      .find('div.article-expansion-panel-container')
      .at(0)
      .simulate('click');

    expect(props.onArticleClick).toBeCalled();
  });

});