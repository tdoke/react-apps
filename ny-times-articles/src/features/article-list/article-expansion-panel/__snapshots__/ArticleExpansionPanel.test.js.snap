// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleExpansionPanel should render ArticleExpansionPanel 1`] = `ShallowWrapper {}`;
